### DISCUSSION TITLE
How to use BaseFinetunning callback and schedulers?

### DISCUSSION THREAD

**Discussion by grudloff** [Lightning-AI/pytorch-lightning#11273] [Lightning Trainer API: Trainer, LightningModule, LightningDataModule]
Is there a way to use the BaseFinetunning callback to dynamically unfreeze layers and assign proportional learning rates to them together with a learning rate scheduler?

Example code I would naively expect to work:

```python
#inside pl_module
    def configure_optimizers(self):
            params = filter(lambda p: p.requires_grad, self.parameters())
            # define optimizer
            # define scheduler
            if scheduler:
                return [optimizer], [scheduler]
            else:
                return optimizer


class BackboneGradualUnfreeze(BaseFinetuning):
        def __init__(self, backbone_epoch=100):
            super().__init__()
            self._backbone_epoch= backbone_epoch

        def freeze_before_training(self, pl_module):
            self.freeze(pl_module.backbone)

        def finetune_function(self, pl_module, current_epoch, optimizer, optimizer_idx):
            if current_epoch == self._backbone_epoch:
                self.unfreeze_and_add_param_group(
                    modules=pl_module.backbone,
                    optimizer=optimizer,
                    initial_denom_lr=1e2
                )

```
